json_tests/tests/test.rs

```diff
@@ -1133,6 +1133,8 @@ fn test_parse_enum_errors() {
             "trailing characters at line 1 column 9"),
         ("\"Frog\"",
             "EOF while parsing a value at line 1 column 6"),
+        ("\"Frog\" 0 ",
+            "EOF while parsing a value at line 1 column 6"),
```

Comment:
I'm not sure about what `ErrorCode` to use. `EOFWhileParsingValue` seemed somewhat appropriate as it keeps the error for message for the `"\"Frog\""` case, but looking at this test, I could also add a new variant. Any preferences?